frame/staking-rewards/src/lib.rs

```diff
@@ -95,11 +99,26 @@ pub mod pallet {
 	#[allow(dead_code)]
 	pub(crate) type PenaltyOf<T> = Penalty<AccountIdOf<T>>;
 
+	// NOTE:
+	// some state processed by batched brain, as these just merged into main collections
+	// some states are processed by fold, because updates happens in place
+	// fold can be made drain if to use A/B storages and drain one and fill other depending on epoch
```

Comment:
Any reference to what A/B storages means?

Reply:
it is like having 2 bottles. one is empty and one is full. one bottle is added with some a little, but than at some point other bottled fully drained into bottle which was added a few. and they switch position. will document that in PR. this is hack for blockhain not having real concurent queues.

Reply:
like A/B deploy. you have A or B.

Reply:
and switch